File: Daily Challenge/February 2025/Clear Digits.cpp
public class Solution {
  public bool IsDigit(char c) {
    return c >= '0' && c <= '9';
  }

  public string ClearDigits(string s) {
    int n = s.Length;
    int countDight = 0;
    string result = "";

    for (int i = n - 1; i >= 0; --i) {
      if (IsDigit(s[i])) {
        countDight++;
      } else {
        if (countDight == 0) {
          result += s[i];
        } else {
          countDight--;
        }
      }
    }

    string finalResult = new string(result.Reverse().ToArray());

    return finalResult;
  }
}

File: Daily Challenge/February 2025/Clear Digits.cs
class Solution {
  bool isdigit(char c) {
    return c >= '0' && c <= '9';
  }

public:
  string clearDigits(string s) {
    int n = s.size();
    int countDigits = 0;

    string result = "";
    for (int i = n - 1; i >= 0; --i) {
      if (isdigit(s[i])) {
        countDigits++;
      } else {
        if (countDigits == 0) {
          result += s[i];
        } else {
          countDigits--;
        }
      }
    }

    reverse(result.begin(), result.end());

    return result;
  }
};